pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -726,5 +726,21 @@ func (p *Provisioner) restoreClusterBackup(cluster *v3.Cluster, spec v3.ClusterS
 	if backup.Spec.ClusterID != cluster.Name {
 		return "", "", "", fmt.Errorf("snapshot [%s] is not a backup of cluster [%s]", backup.Name, cluster.Name)
 	}
-	return p.driverRestore(cluster, spec)
+
+	api, token, cert, err = p.driverRestore(cluster, spec)
+	if err != nil {
+		return "", "", "", err
+	}
+	// checking if we have s3 config and that it's not inconsistent. This happens
+	// when restore is performed with invalid credentials and then the cluster is updated to fix it.
+	if spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig != nil {
```

Comment:
wouldn't error be not nil in this case? `when restore is performed with invalid credentials and then the cluster is updated to fix it.`

Reply:
If the credentials are invalid, err will be nil. Once the cluster is updated with the correct credentials, the provisioner will kickin again, but we will still land in the restore case, not the update case (line 285). 

In this case the restore will complete successfully and error will be nil. But we still need to update the cluster to correct the credentials in the cluster store used by KE.